pkg/handler/gm_config_test.go

```diff
@@ -0,0 +1,178 @@
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			w := httptest.NewRecorder()
+			ctx, _ := gin.CreateTestContext(w)
+			ctx.Request = httptest.NewRequest(http.MethodGet, fmt.Sprintf("/api/v1/configs/gm?%s", tt.query), nil)
+
+			h.GetGmConfig(ctx)
+			require.Equal(t, tt.wantCode, w.Code)
+			expRespRaw, err := ioutil.ReadFile(tt.wantResponsePath)
+			require.NoError(t, err)
+
+			require.JSONEq(t, string(expRespRaw), w.Body.String(), "[Handler.GetGmConfig] response mismatched")
+		})
+	}
+}
+func TestHandler_UpsertGmConfig(t *testing.T) {
+	db := testhelper.LoadTestDB("../../migrations/test_seed")
+	repo := pg.NewRepo(db)
+	cfg := config.LoadTestConfig()
+	log := logger.NewLogrusLogger()
+	entity := entities.New(cfg, log, repo, nil, nil, nil, nil, nil, nil, nil, nil)
+
+	h := &Handler{
+		entities: entity,
+		log:      log,
+	}
+
+	tests := []struct {
+		name             string
+		args             interface{}
+		wantCode         int
+		wantResponsePath string
+	}{
+		{
+			name: "Upsert Gm config channel",
+			args: request.UpsertGmConfigRequest{
+				GuildID:   "552427722551459840",
+				ChannelID: "701029345795375114",
+			},
+			wantCode:         http.StatusOK,
+			wantResponsePath: "testdata/200-message-ok.json",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			body, err := json.Marshal(tt.args)
+			if err != nil {
+				t.Error(err)
+				return
+			}
+
+			w := httptest.NewRecorder()
+			ctx, _ := gin.CreateTestContext(w)
+			ctx.Request = httptest.NewRequest(http.MethodPost, "/api/v1/configs/gm", bytes.NewBuffer(body))
+
+			h.UpsertGmConfig(ctx)
+			require.Equal(t, tt.wantCode, w.Code)
+			expRespRaw, err := ioutil.ReadFile(tt.wantResponsePath)
+			require.NoError(t, err)
+
+			require.JSONEq(t, string(expRespRaw), w.Body.String(), "[Handler.UpsertGmConfig] response mismatched")
+		})
+	}
+}
+
+func TestHandler_GetUserCurrentGMStreak(t *testing.T) {
+	db := testhelper.LoadTestDB("../../migrations/test_seed")
+	repo := pg.NewRepo(db)
+	cfg := config.LoadTestConfig()
+	log := logger.NewLogrusLogger()
+	entity := entities.New(cfg, log, repo, nil, nil, nil, nil, nil, nil, nil, nil)
+
+	h := &Handler{
+		entities: entity,
+		log:      log,
+	}
+
+	tests := []struct {
+		name             string
+		query            string
+		wantCode         int
+		wantResponsePath string
+	}{
+		{
+			name:             "User has Gm streak",
+			query:            "guild_id=552427722551459840&discord_id=393034938028392449",
+			wantCode:         http.StatusOK,
+			wantResponsePath: "testdata/users/200-exist-gm-streak.json",
+		},
+		{
+			name:             "400_empty_guild_id",
+			wantCode:         http.StatusBadRequest,
+			wantResponsePath: "testdata/400-missing-discordID-guildID.json",
+		},
+		{
+			name:             "400_empty_discord_id",
+			wantCode:         http.StatusBadRequest,
+			wantResponsePath: "testdata/400-missing-discordID-guildID.json",
+		},
```

Comment:
missing guild_id in query

Reply:
Currently api returns both if missing one "discord_id and guild_id are required"